feat(metrics-operator): bootstrap AnalysisDefinition (#1823)
keptn/lifecycle-toolkit: docs/content/en/docs/crd-ref/metrics/v1alpha3/_index.md

| `items` _[AnalysisDefinition](#analysisdefinition) array_ |  |

#### AnalysisDefinitionSpec

AnalysisDefinitionSpec defines the desired state of AnalysisDefinition

_Appears in:_
- [AnalysisDefinition](#analysisdefinition)

| Field | Description |
| --- | --- |
| `objectives` _[Objective](#objective) array_ | Objectives defines a list of objectives for evaluation |
| `totalScore` _[Score](#score)_ | TotalScore defines the required score for an evaluation to be successful |

#### Criteria

Criteria defines list of targets for evaluation

_Appears in:_
- [CriteriaSet](#criteriaset)

| Field | Description |
| --- | --- |
| `anyOf` _[Target](#target) array_ | AnyOf contains a list of criteria where any of them needs to be successful for the CriteriaSet to pass |
| `allOf` _[Target](#target) array_ | AllOf contains a list of criteria where all of them need to be successful for the CriteriaSet to pass |

#### CriteriaSet

CriteriaSet represents the set of evaluation criterias

_Appears in:_
- [SLOTarget](#slotarget)

| Field | Description |
| --- | --- |

> | `anyOf` _[Criteria](#criteria) array_ | AnyOf contains a list of targets where any of them needs to be successful for the Criteria to pass |

Reviewer: ```suggestion
| `anyOf` _[Criteria](#criteria) array_ | AnyOf contains a list of targets [t1,t2 ...] where the pass criteria is given only if all target pass ( logical and of all targets)  |
```

> | `allOf` _[Criteria](#criteria) array_ | AllOf contains a list of targets where all of them need to be successful for the Criteria to pass |

Reviewer: maybe here too

| `secretKeyRef` _[SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#secretkeyselector-v1-core)_ | SecretKeyRef defines an optional secret for access credentials to the metrics provider. |

#### ObjectReference

_Appears in:_
- [Objective](#objective)

| Field | Description |
| --- | --- |
| `name` _string_ |  |
| `namespace` _string_ |  |

#### Objective

Objective defines an objective for analysis

_Appears in:_
- [AnalysisDefinitionSpec](#analysisdefinitionspec)

| Field | Description |
| --- | --- |
| `analysisValueTemplateRef` _[ObjectReference](#objectreference)_ | AnalysisValueTemplateRef defines a reference to the used AnalysisValueTemplate |
| `target` _[Target](#target)_ | Target defines failure or warning criteria |
| `weight` _integer_ | Weight can be used to emphasize the importance of one Objective over the others |

> | `keyObjective` _boolean_ | KeyObjective defines if the objective fails when the target is not met |

Reviewer 2: ```suggestion
| `keyObjective` _boolean_ | If KeyObjective is true, the objective fails when the target is not met |
```

Author: We need to start the comment with the name of the parameter we are describing, this is required in technical documentations

#### Operator

Operator specifies the supported operators for value comparisons

_Appears in:_
- [Target](#target)

| Field | Description |
| --- | --- |
| `lessThanOrEqual` _[OperatorValue](#operatorvalue)_ | LessThanOrEqual represents '<=' operator |
| `lessThan` _[OperatorValue](#operatorvalue)_ | LessThan represents '<' operator |
| `greaterThan` _[OperatorValue](#operatorvalue)_ | GreaterThan represents '>' operator |
| `greaterThanOrEqual` _[OperatorValue](#operatorvalue)_ | GreaterThanOrEqual represents '>=' operator |
| `equalTo` _[OperatorValue](#operatorvalue)_ | EqualTo represents '==' operator |

#### OperatorValue

> OperatorValue represents the value the result will be compared to

Reviewer 2: ```suggestion
OperatorValue represents the value to which the result is compared
```

> | `aggregation` _string_ | Aggregation defines as the type of aggregation function to be applied on the data. Accepted values: p90, p95, p99, max, min, avg, median |

Reviewer 2: ```suggestion
| `aggregation` _string_ | Aggregation defines the type of aggregation function to be applied on the data. Accepted values: p90, p95, p99, max, min, avg, median |
```
